src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.cs

```diff
@@ -205,7 +205,7 @@ public async Task GetAsync_IPBasedUri_Success(IPAddress address)
 
             if (PlatformDetection.IsAndroid && options.UseSsl && address == IPAddress.IPv6Loopback)
             {
-                throw new SkipTestException("IPv6 loopback with SSL doesn't work on Android");
+                throw new SkipTestException("TargetHost cannot be set to an IPv6 address on Android because the string doesn't conform to the STD 3 ASCII rules");
```

Comment:
It seems like that can be turned off..???? 
https://developer.android.com/reference/java/net/IDN
Is this because it does not like "::1" format?  
Would it work in the long form? e.g. "0:0:0:0:0:0:0:1" or "[0:0:0:0:0:0:0:1]"?

Reply:
It has an allow list of characters that doesn't include `:`, it's not just about the loopback address specifically so the long forms don't work either. We can't turn it off AFAIK. I'm by no means an expert in this area but from what I read it's a bad practice to put IP addresses into SNI host name (see https://github.com/dotnet/runtime/issues/79143).